.github/workflows/codeql.yml

```diff
@@ -0,0 +1,92 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: CodeQL
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches:
+      - master
+    paths-ignore:
+      - '**/*.md'
+      - '**/*.txt'
+      - '**/*.po'
+  schedule:
```

Comment:
So we run this job:
- on each new commit on master
- on each new commit on a PR
- on master every Wednesday at 4:20

Isn't it redundant ? Should we only keep the "every week on master" policy instead ? (but then what happen if a codesmell is detected ? do we get a warning somewhere ? should we add a tweak here to open an issue automatically ?)

(I personally think the weekly strategy is good enough, our per-commit CI workflow contains already too much stuff and this codesmell thing is really similar in philosophy to dependabot which display warning without blocking the CI)

Reply:
We want to check:

- If a PR don't introduce new security vulnerabilities
- Periodically on master for new security vulnerabilities that could be discovered 
- On a new commit of master because it's the final code once merged (but could be change if we have enabled `merge queue`)

The warning are located https://github.com/Scille/parsec-cloud/security/code-scanning
+ On PR you have

![image](https://user-images.githubusercontent.com/30697622/210242659-b5672f73-69a5-4e0c-aa8f-2f491c9a65ff.png)